okhttp-testing-support/src/main/java/okhttp3/TestUtil.java

```diff
@@ -83,4 +85,9 @@ public static void awaitGarbageCollection() throws Exception {
     Thread.sleep(100);
     System.runFinalization();
   }
+
+  public static void ensureAllConnectionsReleased(OkHttpClient client) {
+    client.connectionPool().evictAll();
+    assertEquals(0, client.connectionPool().idleConnectionCount());
```

Comment:
How does it fail when there is a bug causing the original test to fail, and also retaining connections?  Does this hide the root cause?

Reply:
> How does it fail when there is a bug causing the original test to fail, and also retaining connections?

If there is a failed assertion inside a test and the test also retains connections, the test output will show both the assertion inside the test and the leak assertion.
Without the change, the output shows the failed assertion inside the test and, sometimes a few minutes later, a leak assertion, when the first test that checks for leaks is executed.

> Does this hide the root cause?

I don't think it does.